quantumblacklabs/kedro-viz
# Update Kedro-UI to v1.1.4 & implement relevant changes

## Description

I've added 3 new minor features to [Kedro-UI](https://github.com/quantumblacklabs/kedro-ui) in versions 1.1.3 and 1.1.4, that will allow us to implement or improve some new/existing features:

1. [Replace check-font-loaded](https://github.com/quantumblacklabs/kedro-ui/pull/76) with [Google Webfont Loader](https://github.com/typekit/webfontloader). This will replace [a weird and unreliable bit of the codebase](https://github.com/quantumblacklabs/kedro-viz/blob/main/src/actions/check-font-loaded.js) with a proper callback, which is all we ever really wanted to begin with.
2. [Add transition-out for closing Dropdown](https://github.com/quantumblacklabs/kedro-ui/pull/78). No changes required, other than to update the version.
3. Add active pipeline menu-option border-left colour. This uses [the new className prop](https://github.com/quantumblacklabs/kedro-ui/pull/80).
4. Use [Dropdown disabled prop](https://github.com/quantumblacklabs/kedro-ui/pull/79) for pipeline selector

## Development notes

I've deleted check-font-loaded.js and its tests, as it's no longer needed. This should improve cross-platform stability. I've checked the load-time in Chrome and it appears to be about the same.

## QA notes

1. Test loading the webfont with a slower connection.
2. Try closing the dropdown
3. Try changing the active pipeline and observe the new blue left border on the menu-options, on both themes.
3. See Dropdown when disabled - e.g. when loading, or when there are no pipelines in the dataset. It should no longer be keyboard-navigable.

## Checklist


## Legal notice


- I submit this contribution under the [Apache 2.0 license](https://www.apache.org/licenses/LICENSE-2.0.txt) and represent that I am entitled to do so on behalf of myself, my employer, or relevant third parties, as applicable.
- I certify that (a) this contribution is my original creation and / or (b) to the extent it is not my original creation, I am authorised to submit this contribution on behalf of the original creator(s) or their licensees.
- I certify that the use of this contribution as authorised by the Apache 2.0 license does not violate the intellectual property rights of anyone else.
